### PR TITLE
Test framework related refactors

### PR DESCRIPTION
**Pull request by phillipm**
Pull test refactors out of https://github.com/dimagi/commcare-odk/pull/1059 into a smaller PR.

 - Make `CommCareTestApplication` override session service setup.
 - Dispatch http request factory via (test) application for ease of overriding in test suite (https://github.com/dimagi/commcare-odk/commit/4c1cd5d8264e07a2eaf5f21a25b841bb6d86909a#diff-5ce2bf85747c398fb9c88402a8a693b2R96)
 - Rename app install / login test methods
 - Make tests use `CommCareTestApplication`

Unrelated to tests:
 - Make form submission / sync report through `ConnectorWithResultCallback`
 - Add `hideTaskCancelButton` to `CommCareTaskConnector` interface so that `FormAndDataSyncer.syncData` can be decoupled from `CommCareHomeActivity` (well, really `CommCareActivity`...)

### CODE REVIEW

File: app/src/org/commcare/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -1160,12 +1162,15 @@ public void onClick(DialogInterface dialog, int i) {
...
+    @Override
+    public void reportSuccess(String message) {
+        displayMessage(message, false, false);
     }
 
-    void displayMessage(String message, boolean bad) {
-        displayMessage(message, bad, false);
+
+    @Override
+    public void reportFailure(String message, boolean showPopupNotification) {
```
Comment (wpride) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: Looks like showPopupNotification is always true when we call this - safe to remove?

File: app/src/org/commcare/activities/CommCareHomeActivity.java [hunk 2]
```diff
@@ -1433,76 +1438,16 @@ public void handlePullTaskResult(ResultAndError<DataPullTask.PullTaskResult> res
...
-        if (userTriggeredSync) {
-            GoogleAnalyticsUtils.reportSyncAttempt(
-                    GoogleAnalyticsFields.ACTION_USER_SYNC_ATTEMPT,
-                    reportSyncLabel, reportSyncValue);
-        } else {
-            GoogleAnalyticsUtils.reportSyncAttempt(
-                    GoogleAnalyticsFields.ACTION_AUTO_SYNC_ATTEMPT,
-                    reportSyncLabel, reportSyncValue);
-        }
-        //TODO: What if the user info was updated?
```
Comment (wpride) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: looks like we lost this TODO (not sure if intentional)

File: app/src/org/commcare/activities/SyncUIHandling.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.commcare.tasks.DataPullTask;
+import org.commcare.tasks.ResultAndError;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Default sync task result handling logic.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class SyncUIHandling {
```
Comment (wpride) [inline app/src/org/commcare/activities/SyncUIHandling.java]: OOC do you think this class and the ConnectorWithResultCallback will be used outside of the home activity?

Comment (phillipm) [inline app/src/org/commcare/activities/SyncUIHandling.java]: Yeah, this PR is build up for https://github.com/dimagi/commcare-odk/pull/1059

Comment (wpride) [inline app/src/org/commcare/activities/SyncUIHandling.java]: nice

File: app/src/org/commcare/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (phillipm) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: It is actually used in https://github.com/dimagi/commcare-odk/pull/1059 in some changes that came up during resolving a merge conflict that maybe should have been pulled into here as well...

File: app/src/org/commcare/activities/CommCareHomeActivity.java (hunk 2 above)
Comment (phillipm) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: I don't really understand this TODO but suspect you may have fixed this issue in https://github.com/dimagi/commcare-odk/pull/814

Comment (wpride) [inline app/src/org/commcare/activities/CommCareHomeActivity.java]: nice, sweet fixing by me

**Comment by amstone326** [comment]
@phillipm Has conflicts
@wpride Can you finish up reviewing this once it's mergeable? I'd rather not start on #1059 till this is merged.

**Comment by wpride** [comment]
this is gtm from me once conflicts addressed

**Pull request merged**